library/Icinga/Application/Hook/Common/DbMigration.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/* Icinga Web 2 | (c) 2023 Icinga GmbH | GPLv2+ */
+
+namespace Icinga\Application\Hook\Common;
+
+use ipl\Sql\Connection;
+use RuntimeException;
+
+class DbMigration
+{
+    /** @var string The sql string to be executed */
+    protected $query;
+
+    /** @var string The sql script version the queries are loaded from */
+    protected $version;
+
+    /** @var string */
+    protected $scriptPath;
+
+    /** @var ?string */
+    protected $description;
+
+    /** @var ?string */
+    protected $lastState;
+
+    public function __construct(string $version, string $scriptPath)
+    {
+        $this->scriptPath = $scriptPath;
+
+        $this->setVersion($version);
+    }
+
+    /**
+     * Get the sql script version the queries are loaded from
+     *
+     * @return string
+     */
+    public function getVersion(): string
+    {
+        return $this->version;
+    }
+
+    /**
+     * Set the sql script version the queries are loaded from
+     *
+     * @param string $version
+     *
+     * @return $this
+     */
+    public function setVersion(string $version): self
+    {
+        $this->version = $version;
+
+        return $this;
+    }
+
+    /**
+     * Get upgrade script relative path name
+     *
+     * @return string
+     */
+    public function getScriptPath(): string
+    {
+        return $this->scriptPath;
+    }
+
+    /**
+     * Get the description of this database migration if any
+     *
+     * @return ?string
+     */
+    public function getDescription(): ?string
+    {
+        return $this->description;
+    }
+
+    /**
+     * Set the description of this database migration
+     *
+     * @param ?string $description
+     *
+     * @return DbMigration
+     */
+    public function setDescription(?string $description): self
+    {
+        $this->description = $description;
+
+        return $this;
+    }
+
+    /**
+     * Get the last error message of this hook if any
+     *
+     * @return ?string
+     */
+    public function getLastState(): ?string
+    {
+        return $this->lastState;
+    }
+
+    /**
+     * Set the last error message
+     *
+     * @param ?string $message
+     *
+     * @return $this
+     */
+    public function setLastState(?string $message): self
+    {
+        $this->lastState = $message;
+
+        return $this;
+    }
+
+    /**
+     * Perform the sql migration
+     *
+     * @param Connection $conn
+     *
+     * @return $this
+     *
+     * @throws RuntimeException Throws an error in case of any database errors or when there is nothing to migrate
+     */
+    public function apply(Connection $conn): self
+    {
+        if (! $this->query) {
+            $statements = @file_get_contents($this->getScriptPath());
```

Comment:
> but I see it more and more often that side-effect free getters/setters are called inside a class

Yes. Because we do that since ages. The same reason why something else was *fine* just a few days ago. Please let's discuss this before requiring this generally.

Reply:
Sure. I was just wondering if this is a personal preference or something we encourage. Because I don't remember any PRs with change requests on this.

If you are referring to the `@return void` topic, it is not because we have been leaving it out for ages, but because it is superfluous.